api/trace/testtrace/propagator_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testtrace_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/propagation"
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+type outOfThinAirPropagator struct {
+	t *testing.T
+}
+
+var _ propagation.HTTPPropagator = outOfThinAirPropagator{}
+
+func (p outOfThinAirPropagator) Extract(ctx context.Context, supplier propagation.HTTPSupplier) context.Context {
+	traceID, err := core.TraceIDFromHex("938753245abe987f098c0987a9873987")
+	require.NoError(p.t, err)
+	spanID, err := core.SpanIDFromHex("2345f98c0987a09d")
+	require.NoError(p.t, err)
+	sc := core.SpanContext{
+		TraceID:    traceID,
+		SpanID:     spanID,
+		TraceFlags: 0,
+	}
+	require.True(p.t, sc.IsValid())
+	return trace.ContextWithRemoteSpanContext(ctx, sc)
+}
+
+func (outOfThinAirPropagator) Inject(context.Context, propagation.HTTPSupplier) {}
+
+func (outOfThinAirPropagator) GetAllKeys() []string {
+	return nil
+}
+
+type nilSupplier struct{}
+
+var _ propagation.HTTPSupplier = nilSupplier{}
+
+func (nilSupplier) Get(key string) string {
+	return ""
+}
+
+func (nilSupplier) Set(key string, value string) {}
+
+func TestMultiplePropagators(t *testing.T) {
+	ootaProp := outOfThinAirPropagator{t: t}
+	ns := nilSupplier{}
+	testProps := []propagation.HTTPPropagator{
+		trace.TraceContext{},
+		trace.B3{SingleHeader: false},
+		trace.B3{SingleHeader: true},
+	}
+	bg := context.Background()
+	// sanity check of oota propagator
+	{
+		props := propagation.New(propagation.WithExtractors(ootaProp))
+		ctx := propagation.ExtractHTTP(bg, props, ns)
+		sc := trace.RemoteSpanContextFromContext(ctx)
+		require.True(t, sc.IsValid(), "oota prop failed sanity check")
+	}
+	// sanity check for real propagators
+	for _, prop := range testProps {
+		props := propagation.New(propagation.WithExtractors(prop))
+		ctx := propagation.ExtractHTTP(bg, props, ns)
+		sc := trace.RemoteSpanContextFromContext(ctx)
+		require.Falsef(t, sc.IsValid(), "%#v failed sanity check")
```

Comment:
Missing a format field value.
```suggestion
		require.Falsef(t, sc.IsValid(), "%#v failed sanity check", props)
```

Reply:
I suspect @MrAlias meant `prop` there for that argument.

Reply:
derp, :facepalm:, yes `prop`.

Reply:
Thanks for spotting, the linters we use are not clever enough to spot the formatting issues on functions outside the `fmt` package.